src/geometry/manifold/charted_space.lean

```diff
@@ -554,6 +554,28 @@ begin
   exact h₂.image_of_continuous_on ((chart_at H x).continuous_on_symm.mono h₃)
 end
 
+/-- If a topological space admits an atlas with locally connected charts, then the space itself is
+locally connected. -/
+lemma charted_space.locally_connected_space [locally_connected_space H] :
+  locally_connected_space M :=
+{ open_connected_basis := λ x, begin
+    let e : local_homeomorph M H := chart_at H x,
+    have h : ∀ t : set M, is_open t ∧ x ∈ t ∧ is_connected t → t ∈ 𝓝 x,
+    { rintros t ⟨ht₁, ht₂, -⟩,
+      exact ht₁.mem_nhds ht₂, },
+    rw ← e.symm_map_nhds_eq (mem_chart_source H x) at ⊢ h,
+    have h' : (𝓝 (e x)).has_basis (λ s, (is_open s ∧ e x ∈ s ∧ is_connected s) ∧ s ⊆ e.target) id,
+    { exact (locally_connected_space.open_connected_basis (e x)).restrict_subset
+        (e.open_target.mem_nhds (mem_chart_target H x)) },
+    refine (h'.map e.symm).to_has_basis' _ h,
+    rintros s ⟨⟨hs₁, hs₂, hs₃⟩, hs₄ : s ⊆ _⟩,
+    have hs₅ : e.symm '' s = e.source ∩ e ⁻¹' s := e.symm_image_eq_source_inter_preimage hs₄,
+    refine ⟨e.source ∩ e ⁻¹' s, ⟨_, ⟨mem_chart_source H x, hs₂⟩, _⟩, hs₅.ge⟩,
+    { exact e.preimage_open_of_open hs₁ },
+    { rw ← hs₅,
+      exact hs₃.image e.symm (e.symm.continuous_on.mono hs₄) },
+  end }
```

Comment:
```suggestion
lemma charted_space.locally_compact [locally_compact_space H] : locally_compact_space M :=
begin
  have : ∀ (x : M), (𝓝 x).has_basis
      (λ s, s ∈ 𝓝 (chart_at H x x) ∧ is_compact s ∧ s ⊆ (chart_at H x).target)
      (λ s, (chart_at H x).symm '' s),
  { intro x,
    rw [← (chart_at H x).symm_map_nhds_eq (mem_chart_source H x)],
    exact ((compact_basis_nhds (chart_at H x x)).has_basis_self_subset
      (chart_target_mem_nhds H x)).map _ },
  refine locally_compact_space_of_has_basis this _,
  rintro x s ⟨h₁, h₂, h₃⟩,
  exact h₂.image_of_continuous_on ((chart_at H x).continuous_on_symm.mono h₃)
end
```
or maybe more readable:
```
lemma charted_space.locally_connected_space [locally_connected_space H] :
  locally_connected_space M :=
begin
  let E : M → local_homeomorph M H := λ x, chart_at H x,
  have h : ∀ x, ∀ s : set H, (is_open s ∧ (E x x) ∈ s ∧ is_connected s) ∧ s ⊆ (E x).target →
    is_preconnected ((E x).symm '' s) :=
  λ x s ⟨⟨_, _, hsconn⟩, hssubset⟩,
    hsconn.is_preconnected.image _ ((E x).continuous_on_symm.mono hssubset),
  have h' : ∀ x, (map (E x).symm (𝓝 (E x x))).has_basis
    (λ s, (is_open s ∧ E x x ∈ s ∧ is_connected s) ∧ s ⊆ (E x).target)
    (λ s, (E x).symm '' s) :=
  λ x, ((locally_connected_space.open_connected_basis (E x x)).restrict_subset
    ((E x).open_target.mem_nhds (mem_chart_target H x))).map ((E x).symm),
  simp only [local_homeomorph.symm_map_nhds_eq, mem_chart_source, id.def] at h',
  exact locally_connected_space_of_connected_bases _ _ h' h
end
```

Reply:
Thanks!  I ended up taking something halfway between these two versions, but again exploiting the lemma `locally_connected_space_of_connected_bases` (thank you for thinking ahead to make this nice piece of API).